### PR TITLE
IScriptable::ExecuteMethod(..)

### PR DESCRIPTION
**Pull request by PixelRick**

### CODE REVIEW

File: include/RED4ext/Scripting/IScriptable.hpp [hunk 1]
```diff
@@ -15,9 +19,30 @@ struct IScriptable : ISerializable
...
-    int64_t unk30;
-    int64_t unk38;
+    template <typename ReturnT, typename... Args>
+    ReturnT ExecuteMethod(CName aFunc, Args&&... aArgs)
+    {
+        ReturnT ret {};
+        StackArgs_t args;
+        ((args.emplace_back(nullptr, &aArgs)), ...);
+        CClass* cls = scriptCls ? scriptCls : static_cast<CClass*>(GetType());
+        ExecuteFunction(this, cls->GetFunction(aFunc), &ret, args);
```
Comment (WopsS) [inline include/RED4ext/Scripting/IScriptable.hpp:29]: Function might be null.

File: include/RED4ext/Scripting/IScriptable.hpp [hunk 2]
```diff
@@ -15,9 +19,30 @@ struct IScriptable : ISerializable
...
+        return ret;
+    }
+
+    template<typename... Args>
+    void ExecuteMethod(CName aFunc, Args&&... aArgs)
+    {
+        StackArgs_t args;
+        ((args.emplace_back(nullptr, &aArgs)), ...);
+        CClass* cls = scriptCls ? scriptCls : static_cast<CClass*>(GetType());
+        ExecuteFunction(this, cls->GetFunction(aFunc), nullptr, args);
```
Comment (WopsS) [inline include/RED4ext/Scripting/IScriptable.hpp:39]: Function might be null.

**Review by WopsS — CHANGES_REQUESTED** [review]

**Comment by PixelRick** [comment]
Don't merge just yet, adding some final changes

File: include/RED4ext/Scripting/IScriptable.hpp [hunk 3]
```diff
@@ -23,9 +26,50 @@ struct IScriptable : ISerializable
...
+
+        CBaseFunction* func = cls->GetFunction(aFunc);
+        if (!func)
+        {
+            return false;
+        }
+
+        StackArgs_t args;
+        ((args.emplace_back(nullptr, &aArgs)), ...);
+        return ::RED4ext::ExecuteFunction(static_cast<void*>(this), func, ret, args);
```
Comment (WopsS) [inline include/RED4ext/Scripting/IScriptable.hpp:69]: `static_cast<ScriptInstance>(this)`

**Review by WopsS — CHANGES_REQUESTED** [review]

File: include/RED4ext/Scripting/IScriptable.hpp [hunk 4]
```diff
@@ -23,9 +26,50 @@ struct IScriptable : ISerializable
...
+    std::optional<ReturnT> ExecuteFunction(CName aFunc, Args&&... aArgs)
+    {
+        ReturnT ret{};
+
+        if (!ExecuteFunctionImpl<ReturnT>(aFunc, &ret, std::forward<Args>(aArgs)...))
+        {
+            return std::nullopt;
+        }
+    
+        return std::optional<ReturnT>{ret};
```
Comment (WopsS) [inline include/RED4ext/Scripting/IScriptable.hpp:39]: `std::make_optional`.

**Review by WopsS — APPROVED** [review]

**Pull request merged**